src/main/java/org/terasology/logic/health/HealthCommands.java

```diff
@@ -72,6 +77,63 @@ public String damageWithType(@Sender EntityRef client, @CommandParam("damageType
             return "Specified damage type does not exist.";
         }
     }
+    @Command(shortDescription = "Damage Resistance", runOnServer = true,
+            requiredPermission = PermissionManager.CHEAT_PERMISSION)
+    public String damageResist(@CommandParam("DamageType") String damageType,@CommandParam("Percentage") float value,@Sender EntityRef clientEntity) {
+        if(value >= 0 && value <= 100) {
+            ClientComponent player = clientEntity.getComponent(ClientComponent.class);
+            DamageResistComponent damageResist;
+            if (!player.character.hasComponent(DamageResistComponent.class)) {
+                damageResist = new DamageResistComponent();
+                damageResist.damageTypes = new HashMap<>();
+                player.character.addComponent(damageResist);
+            } else {
+                damageResist = player.character.getComponent(DamageResistComponent.class);
+            }
+            if (damageType.equals("all")) {
+                damageResist.damageTypes=new HashMap<>();
+                damageResist.damageTypes.put("all",value);
+                if (value == 0)
+                    player.character.removeComponent(DamageResistComponent.class);
+            }
+            else {
+                if (prefabManager.exists(damageType)) {
+                    if (damageResist.damageTypes == null)
+                        damageResist.damageTypes = new HashMap<>();
+                    if (value == 0)
+                    {damageResist.damageTypes.remove(damageType);
+                    }
+                    else
+                    {damageResist.damageTypes.put(damageType, value);
+                    damageResist.damageTypes.remove("all");
+                    }
+                }
+                else
+                    return "Not a valid damage type";
+            }
+                return "Resistance:" + value + "% to " + damageType;
+            }
+        else{ return "accepted values:[0 to 100]";
+        }
```

Comment:
many small style issue:
Normalize bracket wraps plz.
after ```if (...)```  - ```<space>{```
"else" wraps as ```} else {```
code on new line after "{"

I don't sure about tera's rules about one-line if body... possibly it must warped with brackets.

(You can use IDEA formatter for this all (Windows: Ctrl+Alt+L))